Add boost feature flag

oikeme opened the pull request:
Add a feature flag for Boost features on the agency pro dashboard

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

* Add a feature flag for Boost features on the agency pro dashboard

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Use the Horizon link below to ensure that the Boost column is no longer available in the agency pro dashboard and the card in the expandable block his also hidden.
* Ensure that the values in the config files for the existing expandable block and new boost value match for the environment and config file.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

oikeme committed:
Add boost feature flag

Add a feature flag for Boost features on the agency pro dashboard

oikeme committed:
Update jetpack-cloud-horizon.json

Resolve conflict and enable expandable block on Horizon

yashwin commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx:
```diff
@@ -39,7 +41,7 @@ export default function BoostSitePerformance( {
 	const href = `${ siteUrlWithScheme }/wp-admin/admin.php?page=jetpack-boost`;
 
-	return (
+	return isBoostEnabled ? (
```
I think we can remove these changes and handle this in the `SiteExpandedContent` component by doing the below.

Instead of 

```
const defaultColumns = [ 'stats', 'boost', 'backup', 'monitor' ];
```

We can do so that the columns are controlled in one centralized place

```
import { siteColumns } from '../utils';
...
...
const defaultColumns = siteColumns.map( ( { key } ) => key );
```

yashwin replied:
By doing this, when we enable the Boost column, it'll be visible in both places. WDYT?

oikeme replied:
It's a much cleaner solution with things being 1:1 right now and I concur with your thoughts. I've made this update and thanks for pointing this out.

yashwin commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/utils.ts:
```diff
@@ -23,23 +23,29 @@ import type {
+	  ]
+	: [];
+
+const extraColumns: SiteColumns = isExpandedBlockEnabled
```
I think we can rename `extraColumns` as `StatsColumn` now. WDYT?

oikeme replied:
This makes sense to me and I've made the change

oikeme committed:
Update limiting Boost in the dashboard columns and expandable block

Update the extraColumns value to statsColumns as it no longer includes Boost and is only stats and use siteColumns to define the defaultColumns identified for use in the expandable block as this is currently 1:1

yashwin reviewed (approved):
LGTM. Thanks for making the suggested changes.

The Boost column & expandable content are no longer visible in the live link and the config file changes are also looking good.
